pandas/core/strings.py

```diff
@@ -1651,17 +1651,36 @@ def str_translate(arr, table, deletechars=None):
 
 def str_get(arr, i):
     """
+    Extract element from each component at specified position.
+
     Extract element from lists, tuples, or strings in each element in the
     Series/Index.
 
     Parameters
     ----------
     i : int
-        Integer index (location)
+        Position of element to extract.
 
     Returns
     -------
     items : Series/Index of objects
+
+    Examples
+    --------
+    >>> s = pd.Series(["String", (1, 2, 3), ["a", "b", "c"], 123])
+    >>> s
+    0       String
+    1    (1, 2, 3)
+    2    [a, b, c]
+    3          123
+    dtype: object
+
+    >>> s.str.get(1)
+    0      t
+    1      2
+    2      b
+    3    NaN
+    dtype: object
```

Comment:
Looks good. May be we could a dictionary to the `Series` too, and an example indexing by a negative number. I'm not sure, as the example could be too complex for the simplicity of the function. But it'd illustrate the different behaviors.

It'd also help to have an explanation before the example, saying something like "When the value is not an iterable, or a dict where the index is not in the keys, `str.get` returns `NaN`".

If you do it you may encounter this bug: #20671. I'd wait until it's fixed, otherwise you'd have to use an index that is a key in the dictionary, and not illustrate the case when the index is not a key, which raises a `KeyError` right now.

Reply:
Hello @datapythonista! I see that you've fixed the bug (which I tried fixing, but your solution is a lot better). Should I proceed writing the docstring with a dictionary & a negative value?